crates/sui-prover/tests/inputs/loop_invariant/external_invalid_arg_name_macro.move

```diff
@@ -0,0 +1,27 @@
+module 0x42::loop_invariant_external_invalid_arg_name_macro_fail;
+
+use prover::prover::ensures;
+
+macro fun test_loop($n: u64): u128 {
+    let mut s: u128 = 0;
+    let mut i = 0;
+
+    while (i < $n) {
+        i = i + 1;
+        s = s + (i as u128);
+    };
+    s
+}
+
+#[spec_only(loop_inv(target = test_spec))]
+#[ext(no_abort)]
+fun loop_inv(i: u64, n: u64, compare: u128): bool {
+    i <= n && (compare == (i as u128) * ((i as u128) + 1) / 2)
+}
+
+#[spec(prove)]
+fun test_spec(n: u64): u128 {
+    let res = test_loop!(n);
+    ensures(res == (n as u128) * ((n as u128) + 1) / 2);
+    res
+}
```

Comment:
Perhaps add something like
```
fun map_test(v: &vector<u8>) {
  let u = v.map!(|x| 0);
  assert(u.length() == v.length());
}
#[spec_only(loop_inv(target=map_test))]
fun map_loop_inv(i: u64, stop: u64, v: vector<u8>) {
  i <= stop && v.length() == stop - i
}
#[spec(prove)]
fun map_test_spec(v: &vector<u8>) {
  map_test(v);
}
```

Reply:
eventually there should be a loop, maybe in `v.length().do!`